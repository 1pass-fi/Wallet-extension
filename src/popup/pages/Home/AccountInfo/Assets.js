import React, { useEffect, useRef, useState } from 'react'
import { connect, useSelector } from 'react-redux'
// actions
import { loadContent } from 'actions/koi'
import { setIsLoading } from 'actions/loading'
import clsx from 'clsx'
import includes from 'lodash/includes'
import isEmpty from 'lodash/isEmpty'
// components
import NFTCard from 'popup/components/NFTCard'
// selectors
import { getDisplayingAccount } from 'popup/selectors/displayingAccount'

const Assets = ({ assets, loadContent, setIsLoading, currentProviderAddress }) => {
  const [filteredAssets, setFilteredAssets] = useState({})
  const displayingAccount = useSelector(getDisplayingAccount)

  useEffect(() => {
    const handleLoadContent = async () => {
      try {
        setIsLoading(true)
        await loadContent()
        setIsLoading(false)
      } catch (err) {
        setIsLoading(false)
        setError(err.message)
      }
    }

    handleLoadContent()
  }, [currentProviderAddress])

  useEffect(() => {
    let showAssets = []
    showAssets = assets.filter((asset) => displayingAccount.address === asset.owner)
    setFilteredAssets(showAssets)
  }, [displayingAccount, assets])

  const [assetsMinHeight, setAssetsMinHeight] = useState(0)
  const assetsRef = useRef(null)

  useEffect(() => {
    const assetsField = assetsRef.current
    if (assetsField) {
      const scrollHeight = assetsField.scrollHeight
      if (scrollHeight < 200) {
        setAssetsMinHeight(0)
        return
      }

      if (scrollHeight >= 200) {
        setAssetsMinHeight(350)
        return
      }
    }
  }, [filteredAssets])

  return (
    <div
      ref={assetsRef}
      style={{ minHeight: `${clsx(assetsMinHeight)}px` }}
      className="w-full bg-trueGray-100 grid grid-cols-3 gap-5 place-items-center px-3 py-1"
    >
      {filteredAssets[0]?.contents?.map((asset, idx) =>
        !includes(asset.name, 'DID Profile Page') ? <NFTCard key={idx} nft={asset} /> : null
      )}
    </div>
  )
}

const mapStateToProps = (state) => ({
  assets: state.assets
})

export default connect(mapStateToProps, {
  loadContent,
  setIsLoading
})(Assets)
