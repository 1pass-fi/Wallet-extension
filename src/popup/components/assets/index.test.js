import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'

import Assets from '.'


describe('Test for Assets component', () => {
  describe('Render without crashing', () => {
    it('renders correctly', () => {
      const { container } = render(<Assets />)
      expect(container).toMatchSnapshot()
    })
  })
})
