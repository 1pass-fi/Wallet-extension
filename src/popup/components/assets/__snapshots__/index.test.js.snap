// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Assets component Render without crashing renders correctly 1`] = `
<div>
  <div
    class="assets"
  >
    <div
      class="assets-add-more-row"
    >
      <div
        class="assets-plus-icon"
      >
        <svg />
      </div>
    </div>
  </div>
</div>
`;
